torch/csrc/jit/passes/onnx/fold_if_node.cpp

```diff
@@ -0,0 +1,229 @@
+#include <torch/csrc/jit/passes/onnx/fold_if_node.h>
+#include <torch/csrc/jit/passes/onnx/helper.h>
+#include <torch/csrc/jit/passes/onnx/scalar_type_analysis.h>
+#include <torch/torch.h>
+
+#include <c10/util/Optional.h>
+#include <algorithm>
+
+namespace torch {
+namespace jit {
+
+namespace onnx {
+using namespace ::c10::onnx;
+}
+
+static bool conditionIfFold(Node* node) {
+  auto cast_node = node->input()->node();
```

Comment:
a lot of assumptions with `node`, consider adding assertions

Reply:
added the assert